packages/dd-trace/src/gateway/dc_block.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+const dc = require('./dc')
+const shimmer = require('shimmer')
+
+function getActiveChannelPrototype () {
+  const noop = () => {}
+
+  const channel = dc.channel('imstealingurprototype')
+
+  // We have to force the channel to switch to ActiveChannel prototype
+  channel.subscribe(noop)
+
+  const activeChannelPrototype = Object.getPrototypeOf(channel)
+
+  channel.unsubscribe(noop)
+
+  return activeChannelPrototype
+}
+
+function wrappedPublish (data) {
+  let firstError
+
+  for (let i = 0; i < this._subscribers.length; i++) {
+    try {
+      const onMessage = this._subscribers[i]
+      onMessage(data, this.name)
+    } catch (err) {
+      if (firstError === undefined && err instanceof DCBlockingError) {
+        firstError = err
+      } else {
+        process.nextTick(() => {
+          throw err
+        })
+      }
+    }
+  }
+
+  if (firstError) throw firstError
+}
+
+function enable () {
+  shimmer.wrap(getActiveChannelPrototype(), 'publish', (originalPublish) => {
+    return wrappedPublish
+  })
+}
+
+function disable () {
+  shimmer.unwrap(getActiveChannelPrototype(), 'publish')
+}
+
+class DCBlockingError extends Error {
+  constructor (message) {
+    super(message)
```

Comment:
you probably want captureStackTrace in this constructor, no?

Reply:
I'm not really sure. I've tried with different second arguments, and to me to stacktrace that makes the most sense is without captureStackTrace. How would you use it ?

Reply:
I'm removing this file for now as it's not needed yet, we can iterate on this file or the node core merge later